File: Solutions/1710/1710.c
/***************************************************************************************************
 * Copyright: washing
 * FileName: 1710.c
 * Author: washing
 * Version: 1.0
 * Date: 2022-11-15 10:05:52
***************************************************************************************************/


static inline int cmp(const void *pa, const void *pb) {
    return (*(int **)pb)[1] - (*(int **)pa)[1];
}

int maximumUnits(int** boxTypes, int boxTypesSize, int* boxTypesColSize, int truckSize){
    qsort(boxTypes, boxTypesSize, sizeof(int *), cmp);
    int ret = 0;
    int idx = 0;
    while (truckSize != 0 && idx < boxTypesSize) {
        int a = boxTypes[idx][0], b = boxTypes[idx][1];
        idx++;
        if (truckSize > a) {
            truckSize -= a;
            ret += a*b;
        } else {
            ret += truckSize*b;
            truckSize = 0;
        }
    }
    return ret;
}

File: Solutions/1710/1710.py
# -*- coding:utf-8 -*-
# Author: washing
# DateTime: 2022/11/15 11:08
# File: 1710.py
# Desc: 


class Solution:
    def maximumUnits(self, boxTypes: List[List[int]], truckSize: int) -> int:
        boxTypes.sort(key=lambda x: x[1],reverse=True)
        ret = 0
        while truckSize and boxTypes:
            a,b = boxTypes.pop(0)
            if truckSize > a:
                truckSize -= a
                ret += a*b
            else:
                ret += truckSize*b
                truckSize = 0
        return ret
